src/renderer/widgets/Background.cpp

```diff
@@ -6,21 +6,18 @@ CBackground::CBackground(const Vector2D& viewport_, const std::string& resourceI
 }
 
 bool CBackground::draw(const SRenderData& data) {
+    if (!asset)
```

Comment:
actually, I'd check for the resourceID being empty here too. `getAssetByID` does a lookup and if the resourceID is empty why would we lookup

Reply:
Another approach would be to add an additional check in CRenderer::getOrCreateWidgetsFor() to see if the desired file exists and push an empty string ("") as the resource ID if it does not. That would allow checking for an empty resource ID in background render and skip the resource lookup, which is wasteful since the resource will never exist anyway.

I didn't do it that way since I wanted to error check the PNG load to guard not only against missing files but also corrupted files and OOM and such. Cairo has a bunch of error states it can return when loading an image.